### DISCUSSION TITLE
Incorrect types for [@types/chart.js]

### DISCUSSION THREAD

**Discussion by zefir-git** [DefinitelyTyped/DefinitelyTyped#63374] [Issues with a @types package] [answered]
The types for [chartj.s](https://www.chartjs.org/) are incorrect.

Some that I have spotted:
`data.datasets[0].fill` -> type is `string | number | boolean | undefined`. In fact this can also be an object including values like `{target: string | () => string, above: string | () => string, below: string | () => string}`

`options.layout` -> type is optional `Chart.ChartLayoutOptions`. `Chart.ChartLayoutOptions` contains a single optional property: `padding`. In fact, it can contain properties like `autoPadding: boolean`, `paddingTop: number` and so on as well.

`options.scales` -> type is `Chart.ChartScales | Chart.LinearScale | Chart.LogarithmicScale | Chart.TimeScale`. I believe the correct type here should be `Record<string, Chart.ChartScales | Chart.LinearScale | Chart.LogarithmicScale | Chart.TimeScale>` instead, because the usage is:
```ts
options: {
    scales: {
        nameOfScale: { // e.g. "x"
            display: false, // ...
        },
        ...
    }
}
```

Chart.js version 4.0.1.

[a comment by an automated account is not shown]

**Comment by lekoaf** [comment, marked as answer]
These typedefinitions only support up to Chart.js 2.9ish.

Since Chart.js 3 it's written in TypeScript.
